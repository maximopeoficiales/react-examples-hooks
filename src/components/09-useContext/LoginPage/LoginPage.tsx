import { useContext } from "react";
import { UserContext } from "../UserContext";

interface MyProps {}
const defaultProps: MyProps = {};
const LoginPage = (props: MyProps) => {
  props = { ...defaultProps, ...props };

  const { user, setUser } = useContext(UserContext);

  const handlerLogin = () => {
    setUser({
      id: 2323,
      name: "maximo junior",
      age: 22,
    });
  };

  return (
    <div data-testid="LoginPage" className="row justify-content-center">
      <div className="col-md-10">
        <h1>LoginPage component</h1>
        <pre>{JSON.stringify(user, null, 3)}</pre>

        <button
          type="button"
          name=""
          id=""
          onClick={handlerLogin}
          className="btn btn-primary btn-lg btn-block"
        >
          Login
        </button>
      </div>
    </div>
  );
};

export default LoginPage;
