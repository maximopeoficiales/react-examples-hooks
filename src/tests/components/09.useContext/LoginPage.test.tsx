import { act } from "@testing-library/react";
import { mount } from "enzyme";
import React from "react";
import LoginPage from "../../../components/09-useContext/LoginPage/LoginPage";
import {
  User,
  UserContext,
} from "../../../components/09-useContext/UserContext";

describe("Pruebas en LoginPage", () => {
  // mount monta el componente con todos sus hijos
  const setUser = jest.fn();
  const user: User = {
    id: 1,
    age: 33,
    name: "jorge",
  };
  const newUser: User = {
    id: 2323,
    name: "maximo junior",
    age: 22,
  };
  const wrapper = mount(
    <UserContext.Provider value={{ user: {}, setUser }}>
      <LoginPage />
    </UserContext.Provider>
  );

  test("debe de mostrarse correctament", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de ejecutar el setUser con el argumento esperado", () => {
    wrapper.find("button").simulate("click");
    expect(setUser).toBeCalledTimes(1);
    expect(setUser).toHaveBeenCalledWith(newUser);
    // console.log(wrapper.find("pre").html());
    // console.log(wrapper.find("pre").html());
  });
});
