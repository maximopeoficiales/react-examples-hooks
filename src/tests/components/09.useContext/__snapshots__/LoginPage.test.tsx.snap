// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en LoginPage debe de mostrarse correctament 1`] = `
<div
  className="row justify-content-center"
  data-testid="LoginPage"
>
  <div
    className="col-md-10"
  >
    <h1>
      LoginPage component
    </h1>
    <pre>
      {}
    </pre>
    <button
      className="btn btn-primary btn-lg btn-block"
      id=""
      name=""
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
  </div>
</div>
`;
